sonic_y_cable/y_cable.py

```diff
@@ -507,3 +536,384 @@ def check_if_link_is_active_for_torB(physical_port):
+
+    Returns:
+        a boolean, true if the disable is successful
+                 , false if the disable failed
+    """
+
+    buffer = bytearray([target])
+    curr_offset = Y_CABLE_TARGET
+
+    if platform_chassis is not None:
+        result = platform_chassis.get_sfp(
+            physical_port).write_eeprom(curr_offset, 1, buffer)
+        if result is False:
+            return result
+        buffer = bytearray([0])
+        curr_offset = Y_CABLE_ENABLE_LOOPBACK
+        result = platform_chassis.get_sfp(
+            physical_port).write_eeprom(curr_offset, 1, buffer)
+
+    else:
+        helper_logger.log_error("platform_chassis is not loaded, failed to configure the PRBS type")
+        return -1
+
+    return result
+
+
+@hook_y_cable_simulator
+def get_ber_info(physical_port, target):
+    """
+    This API specifically returns the BER (Bit error rate) value for a specfic port.
+    The target could be local side, TOR1, TOR2, NIC etc.
+
+
+    Register Specification of upper page 0x5 at offset 130 is documented below
+
+    Byte offset   bits    Name                     Description
+    130           1-0     Initiate BER Measurement "Write 0x00 - initiate gated BER measurement on target side with PRBS traffic
+                                                    Write 0x01-0xFF - reserved
+                                                    Read 0x00 - BER Gate in process
+                                                    Read 0x01 - BER Gate complete, valid values in Lane BER registers
+                                                    NB this command is only valid when PRBS has been enabled on at least one lane
+                                                    The cable supports 3 modes of operation : mission mode; PRBS mode or loopback mode.
+                                                    Enabling PRBS on any lane of any sides puts cable in PRBS mode and disables loopback and mission mode."
+
+    Args:
+        physical_port:
+             an Integer, the actual physical port connected to a Y cable
+        target:
+             an Integer, the target on which to enable the PRBS
+                         0 -> local side,
+                         1 -> TOR 1
+                         2 -> TOR 2
+                         3 -> NIC
+    Returns:
+        a list, with BER values of lane 0 and lane 1 with corresponding index
+    """
+
+    buffer = bytearray([target])
+    curr_offset = Y_CABLE_TARGET
+
+    ber_result = []
+
+    if platform_chassis is not None:
+        result = platform_chassis.get_sfp(
+            physical_port).write_eeprom(curr_offset, 1, buffer)
+        if result is False:
+            return result
+        buffer = bytearray([0])
+        curr_offset = Y_CABLE_INITIATE_BER_MEASUREMENT
+        result = platform_chassis.get_sfp(
+            physical_port).write_eeprom(curr_offset, 1, buffer)
+        if result is False:
+            return result
+        while(1):
```

Comment:
Suggest `while True:` to be more Pythonic.

Reply:
fixed